build.gradle

```diff
@@ -167,8 +170,35 @@ subprojects {
   }
 }
 
+task codeCoverageReport (type: JacocoReport, group: 'Coverage reports') {
+  description = "Creates an aggregate coverage for the whole project."
+  def reportingProjects = subprojects.findAll()
+  dependsOn(reportingProjects.tasks.collectMany{it.withType(Test)})
+
+  onlyIf = { true } // required to circumvent a wrong check in the jacoco report plugin (v2.13)
+
+  sourceDirectories = files() //must be set
+  classDirectories = files()  //must be set
+  additionalSourceDirs files(reportingProjects.sourceSets.main.allSource.srcDirs)
+  additionalClassDirs files(reportingProjects.sourceSets.main.output)
+
+  executionData = files(reportingProjects.jacocoTestReport.executionData)
+
+  doFirst {
+    // find all *.exec files
+    executionData = fileTree(project.rootDir.absolutePath).include("**/build/jacoco/*.exec")
+  }
+
+  reports {
+    html.enabled = true
+    xml.enabled = true
+    xml.destination "${buildDir}/reports/jacoco/report.xml" // report must be here for codecov to pick it up
+    csv.enabled = false
+  }
+}
+
 if (gradle.startParameter.taskNames == ["travisCiBuild"]) {
-  gradle.startParameter.taskNames = ["build"]
+  gradle.startParameter.taskNames = ["build", "codeCoverageReport"]
```

Comment:
Codecov says that they automatically merge all results, and it does not really cost performance. I'd like to keep the logic simple if we can.